### ISSUE TITLE
Populating a Custom Options dropdown with a function

### ISSUE THREAD

**Issue by mfeingesicht** [WeakAuras/WeakAuras2#2204]
**Is your feature request related to a problem? Please describe.**
Have the possibility to populate dropdowns in Custom Options using a function.

**Describe the solution you'd like**
The possibility to choose a custom function when inputting the values of the dropdown, or the possibility to edit the dropdown from Actions or anywhere else.

**Describe alternatives you've considered**
Hard coding the combinations in On Init Actions is easy but not as intuitive.

**Additional context**
This came up when I tried to make a WA that would pop up if I have the wrong set equipped on a raid boss and would have the possibility of choosing which set to use on which boss in Custom Options by selecting from Dropdown Menus.

**Comment by emptyrivers** [comment]
This is closely related to #1078. The main problem is that the options aren't remotely set up to handle "live" updates like this. A few problems that would need to be considered:

- When to run the custom function to determine what should be displayed to the user? Refreshing the options table is pretty expensive computationally, and we have limited ability to tune that performance. Especially once we're injecting unknown userland code into the pipeline, and running it on <insert events here> on the author's request.
- How to handle an option being lost that the user had configured? What if the value the user chose is no longer acceptable?
- It is a guarantee that the `aura_env.config` table will never be unexpectedly changed (that is, between `init` and any other custom code, `aura_env.config` is always constant). In order to maintain that invariant (which many people rely on), we will be forced to completely refresh the aura, and flag `init` to run again, if we detect that the options provided by custom code have changed. I don't want to do that, for both performance and functionality reasons.

In summary, yes it would be nice for it to magically work out like you wanted, but I don't think it's worth attempting at this point.
